### PR TITLE
feat(url): syncronize url and state with the query

### PR DESCRIPTION
**Pull request by tiborux**
Run the project, and check how the url is updated with the query and in the other way around, if you type a query on the url, xcomponents will be opened and it will be set the new query.

EX-3509

### CODE REVIEW

File: packages/x-components/src/utils/string.ts [hunk 1]
```diff
@@ -29,3 +29,17 @@ export function toKebabCase(str: string): string {
...
     .toLowerCase();
 }
+
+/**
+ * Util to capitalize a string .
+ *
+ * @example
+ * Transforms `query` into `Query`
+ *
+ * @param str - String value.
```
Comment (pmareke) [inline packages/x-components/src/utils/string.ts:39]: Add a blank line below please.

File: packages/x-components/src/utils/string.ts [hunk 2]
```diff
@@ -29,3 +29,17 @@ export function toKebabCase(str: string): string {
...
+ * @returns Returns the string capitalized.
```
Comment (pmareke) [inline packages/x-components/src/utils/string.ts:40]: Add a blank line below please.

File: packages/x-components/src/x-modules/url/components/__tests__/utils.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Store } from 'vuex';
```
Comment (pmareke) [inline packages/x-components/src/x-modules/url/components/__tests__/utils.ts:1]: Remove this file please.

File: packages/x-components/src/x-modules/url/components/__tests__/url-handler.spec.ts [hunk 4]
```diff
@@ -25,9 +27,11 @@ function renderUrlHandler({ template = `<UrlHandler />` }: UrlHandlerOptions = {
       localVue
     }
   );
-
+  const wrapper = wrapperTemplate.findComponent(UrlHandler);
+  const $x = wrapperTemplate.vm.$x;
   return {
```
Comment (pmareke) [inline packages/x-components/src/x-modules/url/components/__tests__/url-handler.spec.ts:32]: Add a new line above and remove constants:

```javascript
  return {
    wrapper: wrapperTemplate.findComponent(UrlHandler),
    $x: wrapperTemplate.vm.$x
  };
```

**Review by pmareke — CHANGES_REQUESTED** [review]
Well done!

**Review by pmareke — APPROVED** [review]

**Review by pmareke — APPROVED** [review]

**Review by guillei10 — APPROVED** [review]

**Pull request merged**